component/common/loki/http/server.go

```diff
@@ -0,0 +1,94 @@
+package http
+
+import (
+	"fmt"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+)
+
+// TargetServer is wrapper around WeaveWorks Server that handled some common configuration used in all flow source
+// components that expose a network server. It just handles configuration and initialization, the handlers implementation
+// are left to the consumer.
+type TargetServer struct {
+	logger           log.Logger
+	config           *server.Config
+	metricsNamespace string
+	server           *server.Server
+}
+
+// NewTargetServer creates a new TargetServer, applying some defaults to the server configuration.
+func NewTargetServer(logger log.Logger, metricsNamespace string, reg prometheus.Registerer, config *ServerConfig) (*TargetServer, error) {
+	if !model.IsValidMetricName(model.LabelValue(metricsNamespace)) {
+		return nil, fmt.Errorf("metrics namespace is not prometheus compatiible: %s", metricsNamespace)
+	}
+
+	t := &TargetServer{
+		logger:           logger,
+		metricsNamespace: metricsNamespace,
+	}
+
+	// convert from River into the weaveworks config
+	serverCfg := config.Convert()
```

Comment:
I'm on the fence about this, even though there's not a super big difference right now. Do we want to call Convert() here, or ask component authors to call the Convert() method and pass you the weaveworks config directly?

Let's keep is as-is for now but keep it in mind in case we want to change it in the future.

Reply:
I like the idea of TargetServer taking directly the River-based configuration, since it's a customized version of the WeaveWorks server. Also, by taking directly the River config we could narrow down the configs we expose, weaveworks model is rather complex.

About the component authors doing the convert, you mean so that they are able to tune some other param our river model doesn't expose?